src/TraceEvent/EventPipe/StackCache.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tracing.EventPipe
+{
+    class StackCache
+    {
+        public bool TryGetStack(int stackId, out int stackBytesSize, out IntPtr stackBytes)
+        {
+            if(_stacks.TryGetValue(stackId, out StackMarker stack))
+            {
+                stackBytesSize = stack.StackBytesSize;
+                stackBytes = stack.StackBytes;
+                return true;
+            }
+            else
+            {
+                stackBytesSize = 0;
+                stackBytes = IntPtr.Zero;
+                return false;
+            }
+        }
+
+        public void ProcessStackBlock(byte[] stackBlock)
+        {
+            PinnedBuffer buffer = new PinnedBuffer(stackBlock);
```

Comment:
Same here - wondering if we can replace the pinned handle with fixed keyword.

Reply:
I think this one is in the same bucket as above. We could use fixed to pin/unpin on every dispatch, but its not clear it would be a perf win and/or if the current scheme is a perf problem. If you think it needs to be investigated let me know, otherwise I wasn't planning to optimize read performance until we had some perf data/customer feedback that said read performance was too slow.